src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -1041,7 +1056,31 @@ public static void AnalyzeTargetHelper(TContext context, IEnumerable<Skimmer<TCo
 
                 try
                 {
+                    Stopwatch stopwatch = context.Traces.HasFlag(DefaultTraces.RuleScanTime)
```

Comment:
>stopwatch [](http://example.com/codeflow?start=30&length=9)

Shaopeng, here's a clean pattern for producing timing data on a per-rule + target file basis. We can also consider adding memory consumption information in a similar change, if helpful.